--- INCLUDE/OWL/DISPATCH.H
+++ include64/OWL/DISPATCH.H
@@ -7,6 +7,7 @@
 //----------------------------------------------------------------------------
 #if !defined(OWL_DISPATCH_H)
 #define OWL_DISPATCH_H
+#define MDBCC_WIN64_OVERLAY_DISPATCH 1
 
 #if !defined(OWL_OWLDEFS_H)
 # include <owl/owldefs.h>
@@ -36,7 +37,7 @@
 // - wParam
 // - lParam
 //
-typedef int32 _OWLFUNC (*TAnyDispatcher)(GENERIC&, TAnyPMF, uint, int32);
+typedef LRESULT _OWLFUNC (*TAnyDispatcher)(GENERIC&, TAnyPMF, uint, LPARAM);
 
 //
 // LEGEND: in order to keep dispatcher names from getting too long, the
@@ -68,216 +69,216 @@
 //
 // passes lParam as an int32 and returns an int result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 i_LPARAM_Dispatch(GENERIC& generic,
-                  int     (GENERIC::*pmf)(int32),
+                  int     (GENERIC::*pmf)(LPARAM),
                   uint     wParam,
-                  int32    lParam);
+                  LPARAM    lParam);
 
 //
 // passes wParam as a uint and returns an int result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 i_WPARAM_Dispatch(GENERIC& generic,
                   int     (GENERIC::*pmf)(uint),
                   uint     wParam,
-                  int32    lParam);
+                  LPARAM    lParam);
 
 //----------------------------------------------------------------------------
 
 //
 // passes nothing and returns an int32 result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 I32_Dispatch(GENERIC& generic,
              uint32  (GENERIC::*pmf)(),
              uint,
-             int32);
+             LPARAM);
 
 //
 // passes lParam as an int32 and returns an int32 result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 I32_LPARAM_Dispatch(GENERIC& generic,
-                    int32   (GENERIC::*pmf)(int32),
+                    int32   (GENERIC::*pmf)(LPARAM),
                     uint,
-                    int32    lParam);
+                    LPARAM    lParam);
 
 //
 // passes lParam as a uint and returns an int32 result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 I32_U_Dispatch(GENERIC& generic,
                int32   (GENERIC::*pmf)(uint),
                uint,
-               int32    lParam);
+               LPARAM    lParam);
 
 //
 // passes wParam as a uint and lParam as an int32 and returns an int32 result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 I32_WPARAM_LPARAM_Dispatch(GENERIC& generic,
-                           int32   (GENERIC::*pmf)(uint, int32),
+                           int32   (GENERIC::*pmf)(uint, LPARAM),
                            uint     wParam,
-                           int32    lParam);
+                           LPARAM    lParam);
 
 //----------------------------------------------------------------------------
 
 //
 // passes no arguments and returns a uint result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_Dispatch(GENERIC& generic,
            uint    (GENERIC::*pmf)(),
            uint     wParam,
-           int32    lParam);
+           LPARAM    lParam);
 
 //
 // passes lParam as an int32 and returns a uint result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_LPARAM_Dispatch(GENERIC& generic,
-                  uint    (GENERIC::*pmf)(int32),
+                  uint    (GENERIC::*pmf)(LPARAM),
                   uint     wParam,
-                  int32    lParam);
+                  LPARAM    lParam);
 
 //
 // passes lParam as a TPoint& and returns a uint result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_POINT_Dispatch(GENERIC& generic,
                  uint    (GENERIC::*pmf)(TPoint&),
                  uint,
-                 int32    lParam);
+                 LPARAM    lParam);
 
 //
 // passes lParam as a void* and returns a uint result 
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_POINTER_Dispatch(GENERIC& generic,
                    uint    (GENERIC::*pmf)(void*),
                    uint,
-                   int32    lParam);
+                   LPARAM    lParam);
 
 //
 // passes lParam as a uint and returns a uint result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_U_Dispatch(GENERIC& generic,
              uint    (GENERIC::*pmf)(uint),
              uint,
-             int32    lParam);
+             LPARAM    lParam);
 
 //
 // passes wParam as a uint, lParam.lo as a uint, and lParam.hi as a uint
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_U_U_U_Dispatch(GENERIC& generic,
                  uint    (GENERIC::*pmf)(uint, uint, uint),
                  uint     wParam,
-                 int32    lParam);
+                 LPARAM    lParam);
 
 //
 // passes wParam as a uint and lParam as an int32 and returns a uint result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 U_WPARAM_LPARAM_Dispatch(GENERIC& generic,
-                         uint    (GENERIC::*pmf)(uint, int32),
+                         uint    (GENERIC::*pmf)(uint, LPARAM),
                          uint     wParam,
-                         int32    lParam);
+                         LPARAM    lParam);
 
 //----------------------------------------------------------------------------
 
 //
 // passes nothing and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_Dispatch(GENERIC& generic,
            void    (GENERIC::*pmf)(),
            uint,
-           int32);
+           LPARAM);
 
 //
 // passes lParam as an int32 and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_LPARAM_Dispatch(GENERIC& generic,
-                  void    (GENERIC::*pmf)(int32),
+                  void    (GENERIC::*pmf)(LPARAM),
                   uint,
-                  int32    lParam);
+                  LPARAM    lParam);
 
 //
 // passes lParam as a TPoint& and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_POINT_Dispatch(GENERIC& generic,
                  void    (GENERIC::*pmf)(TPoint&),
                  uint,
-                 int32    lParam);
+                 LPARAM    lParam);
 
 //
 // passes lParam as a void* and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_POINTER_Dispatch(GENERIC& generic,
                    void    (GENERIC::*pmf)(void*),
                    uint,
-                   int32    lParam);
+                   LPARAM    lParam);
 
 //
 // passes lParam as a uint and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_U_Dispatch(GENERIC& generic,
              void    (GENERIC::*pmf)(uint),
              uint,
-             int32    lParam);
+             LPARAM    lParam);
 
 //
 // passes wParam as a uint and lParam as a TPoint& and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_U_POINT_Dispatch(GENERIC& generic,
                    void    (GENERIC::*pmf)(uint, TPoint&),
                    uint     wParam,
-                   int32    lParam);
+                   LPARAM    lParam);
 
 //
 // passes wParam as a uint and lParam as a uint and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_U_U_Dispatch(GENERIC& generic,
                void    (GENERIC::*pmf)(uint, uint),
                uint     wParam,
-               int32    lParam);
+               LPARAM    lParam);
 
 //
 // passes wParam as a uint, lParam.lo as a uint, and lParam.hi as a uint and
 // always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_U_U_U_Dispatch(GENERIC& generic,
                  void    (GENERIC::*pmf)(uint, uint, uint),
                  uint     wParam,
-                 int32    lParam);
+                 LPARAM    lParam);
 
 //
 // passes wParam as a uint and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_WPARAM_Dispatch(GENERIC& generic,
                   void    (GENERIC::*pmf)(uint),
                   uint     wParam,
-                  int32    lParam);
+                  LPARAM    lParam);
 
 //
 // passes wParam as a uint and lParam as an int32 and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_WPARAM_LPARAM_Dispatch(GENERIC& generic,
-                         void    (GENERIC::*pmf)(uint, int32),
+                         void    (GENERIC::*pmf)(uint, LPARAM),
                          uint     wParam,
-                         int32    lParam);
+                         LPARAM    lParam);
 
 
 //----------------------------------------------------------------------------
@@ -288,22 +289,22 @@
 // 32-bit: wParam.lo, lParam,    wParam.hi
 // 16-bit: wParam,    lParam.lo, lParam.hi
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 i_U_W_U_Dispatch(GENERIC& generic,
                  int     (GENERIC::*pmf)(uint, uint, uint),
                  uint     wParam,
-                 int32    lParam);
+                 LPARAM    lParam);
 
 //
 // passes two uints and a HANDLE and always returns 0
 // 32-bit passes: wParam.lo, wParam.hi, lParam
 // 16-bit passes: wParam,    lParam.lo, lParam.hi (same as v_U_U_U)
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_U_U_W_Dispatch(GENERIC& generic,
                  void    (GENERIC::*pmf)(uint, uint, uint),
                  uint     wParam,
-                 int32    lParam);
+                 LPARAM    lParam);
 
 //----------------------------------------------------------------------------
 // message-specific crackers
@@ -312,42 +313,42 @@
 // cracker for WM_ACTIVATE
 // passes a uint, a bool, and an HWND and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_Activate_Dispatch(GENERIC& generic,
                     void    (GENERIC::*pmf)(uint, uint, uint),
                     uint     wParam,
-                    int32    lParam);
+                    LPARAM    lParam);
 
 //
 // cracker for WM_MDIACTIVATE
 // passes two HWNDs and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_MdiActivate_Dispatch(GENERIC& generic,
                        void    (GENERIC::*pmf)(uint, uint),
                        uint     wParam,
-                       int32    lParam);
+                       LPARAM    lParam);
 
 
 //
 // cracker for WM_PARENTNOTIFY
 // passes two uints and an HWND and returns a 32bit result
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 I32_MenuChar_Dispatch(GENERIC& generic,
                       int32   (GENERIC::*pmf)(uint, uint, uint),
                       uint     wParam,
-                      int32    lParam);
+                      LPARAM    lParam);
 
 //
 // cracker for WM_PARENTNOTIFY
 // passes three uints and always returns 0
 //
-int32 _OWLFUNC
+LRESULT _OWLFUNC
 v_ParentNotify_Dispatch(GENERIC& generic,
                         void    (GENERIC::*pmf)(uint, uint, uint),
                         uint     wParam,
-                        int32    lParam);
+                        LPARAM    lParam);
 
 
 //----------------------------------------------------------------------------
